docs/enhancements/ipxe-host-boot.md

```diff
@@ -0,0 +1,125 @@
+---
+title: ipxe-host-boot
+authors:
+  - "@carbonin"
+  - “@mhrivnak”
+creation-date: 2022-02-22
+last-updated: 2022-02-22
+---
+
+# iPXE Host Boot
+
+## Summary
+
+Not all user infrastructure is able to boot isos directly, and iPXE is a commonly 
+used alternative for booting machines. We should support this by making the 
+application directly serve iPXE artifacts so that systems can boot with no 
+other iPXE infrastructure required.
+
+## Motivation
+
+Booting a host using iPXE should be a first-class operation when using
+assisted-service. This has been found to be at least as common a method 
+as providing the live ISO via virtual media and is also required in some 
+environments. Users are already unpacking the ISO we provide into PXE 
+artifacts and serving those in their infrastructure. We can make this much 
+easier for them by serving those artifacts as well as the iPXE boot script 
+directly from our application.
+
+### Goals
+
+- Serve iPXE artifacts (kernel, rootfs, and initrd)
+- Serve an iPXE boot script
+- Include the ignition in the provided initrd
+
+### Non-Goals
+
+- Support PXE (tftp based)
+- Integration with ironic (separate effort)
+- iPXE boot script customization (maybe in a follow-up if it’s useful)
+- Configure the minimal ISO to pull the rootfs from the image service
```

Comment:
What does minimal ISO have to do with this proposal?

Reply:
Currently when we create the minimal ISO template image we embed a url to the rootfs from the OsImages structure.
Now that we're serving the rootfs we could point back to our own service rather than requiring the user mirror the rootfs in addition to the full iso.

I'm saying doing that is out of scope here but could be something we do in the future, but I can remove it from this list if it's causing confusion.

Reply:
worth having a task on that? It would be more of a self contained solution, right?

Reply:
Can leave it here - let's open a task

Reply:
Created MGMT-9406 which is marked as blocking on MGMT-9383